src/test/ui/rfc-2294-if-let-guard/feature-gate.rs

```diff
@@ -9,9 +9,11 @@ fn _if_let_guard() {
 
         () if (let 0 = 1) => {}
         //~^ ERROR `let` expressions in this position are unstable
+        //~| ERROR expected expression, found `let` statement
```

Comment:
At least for now, parentheses and `||` operators are not supported in a `let_chains` context. MIR construction of things like `if (let Some(_) = a && true) || false` is not a reality yet.